packages/location-field/src/index.js

```diff
@@ -0,0 +1,831 @@
+          <Bus size={13} />
+          <div style={{ fontSize: "8px" }}>
+            {humainzeDistanceStringImperial(stop.dist, true)}
+          </div>
+        </div>
+        <div style={{ marginLeft: "30px" }}>
+          <div>
+            {stop.name} ({stop.code})
+          </div>
+          <div style={{ fontSize: "9px" }}>
+            {(stop.routes || []).map(route => {
+              const name = route.shortName || route.longName;
+              return (
+                <Styled.RouteName key={`route-${name}`}>
+                  {name}
+                </Styled.RouteName>
+              );
+            })}
+          </div>
+        </div>
+        <div style={{ clear: "both" }} />
+      </div>
+    </Styled.MenuItem>
+  );
+}
+
+class LocationField extends Component {
+  geocodeAutocomplete = throttle(1000, text => {
+    if (!text) {
+      console.warn("No text entry provided for geocode autocomplete search.");
+      return;
+    }
+    const { geocoderConfig } = this.props;
+    getGeocoder(geocoderConfig)
+      .autocomplete({ text })
+      .then(result => {
+        this.setState({ geocodedFeatures: result.features });
+      })
+      .catch(err => {
+        console.error(err);
+      });
+  });
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      value:
+        props.location && !props.hideExistingValue ? props.location.name : "",
+      menuVisible: false,
+      geocodedFeatures: [],
+      activeIndex: null
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    // If location is updated externally, replace value and geocoded features
+    // in internal state.
+    // TODO: This might be considered an anti-pattern. There may be a more
+    // effective way to handle this.
+    const { location } = this.props;
+    if (location !== prevProps.location) {
+      /* FIXME only disabled this because it'd take longer to refactor */
+      /* eslint-disable-next-line */
+      this.setState({
+        value: location !== null ? location.name : "",
+        geocodedFeatures: []
+      });
+    }
+  }
+
+  getFormControlClassname() {
+    const { locationType } = this.props;
+    return `${locationType}-form-control`;
+  }
+
+  setLocation(location) {
+    const { onLocationSelected, locationType } = this.props;
+    onLocationSelected({ locationType, location });
+    this.setState({ menuVisible: false });
+  }
+
+  useCurrentLocation = () => {
+    const {
+      currentPosition,
+      getCurrentPosition,
+      onLocationSelected,
+      locationType
+    } = this.props;
+    const location = currentPositionToLocation(currentPosition);
+    if (location) {
+      // If geolocation is successful (i.e., user has granted app geolocation
+      // permission and coords exist), set location.
+      onLocationSelected({ locationType, location });
+    } else {
+      // Call geolocation.getCurrentPosition and set as from/to locationType
+      getCurrentPosition(locationType);
```

Comment:
Does `getCurrentPosition` not take a `onLocationSelected` callback? See https://github.com/opentripplanner/otp-react-redux/compare/component-refactor#diff-b47b2554cff82708da15605b764f0e6bR296-R297

Reply:
Also, when I select 'Use Current Location in storybook: https://ibi-group.github.io/otp-ui/?path=/story/locationfield--locationfield-with-selected-location-in-mobile-context

I don't see the behavior function as it does in otp-rr (i.e., it doesn't seem to trigger a getCurrentPosition action). Perhaps this is because `currentPosition` is always provided? Please add a storybook entry for when currentPosition is not available.